File: source-connector/src/main/java/sourceconnector/domain/processor/AbstractMapperProcessor.java
package sourceconnector.domain.processor;

public abstract class AbstractMapperProcessor<Log> extends BaseProcessor<Log> {

  protected abstract Log map(Log input);

  @Override
  public Log process(Log input) {
    Log result = this.map(input);
    if (this.next != null) {
      return this.next.process(result);
    }
    return result;
  }
}

File: source-connector/src/main/kotlin/sourceconnector/domain/processor/AbstractMapperProcessor.kt
package sourceconnector.domain.processor

abstract class AbstractMapperProcessor<Log> : BaseProcessor<Log>() {
  abstract fun map(input: Log): Log

  override fun process(input: Log): Log? {
    val result = this.map(input)

    if (this.next != null) {
      return this.next!!.process(result)
    }
    return result
  }
}
